e3sm_diags/plot/zonal_mean_xy_plot.py

```diff
@@ -36,67 +36,11 @@
 BORDER_PADDING = (-0.14, -0.06, 0.04, 0.08)
 
 
-def _save_plot(fig: plt.figure, parameter: CoreParameter):
-    """Save the plot using the figure object and parameter configs.
-
-    This function creates the output filename to save the plot. It also
-    saves each individual subplot if the reference name is an empty string ("").
-
-    Parameters
-    ----------
-    fig : plt.figure
-        The plot figure.
-    parameter : CoreParameter
-        The CoreParameter with file configurations.
-    """
-    for f in parameter.output_format:
-        f = f.lower().split(".")[-1]
-        fnm = os.path.join(
-            get_output_dir(parameter.current_set, parameter),
-            parameter.output_file + "." + f,
-        )
-        plt.savefig(fnm)
-        logger.info(f"Plot saved in: {fnm}")
-
-    # Save individual subplots
-    if parameter.ref_name == "":
-        PANELs = [TWO_PANELS[0]]
-    else:
-        PANELs = TWO_PANELS
-
-    for f in parameter.output_format_subplot:
-        fnm = os.path.join(
-            get_output_dir(parameter.current_set, parameter),
-            parameter.output_file,
-        )
-        page = fig.get_size_inches()
-
-        for idx, PANEL in enumerate(PANELs):
-            # Extent of subplot
-            subpage = np.array(PANEL).reshape(2, 2)
-            subpage[1, :] = subpage[0, :] + subpage[1, :]
-            subpage = subpage + np.array(BORDER_PADDING).reshape(2, 2)
-            subpage = list(((subpage) * page).flatten())  # type: ignore
-            extent = Bbox.from_extents(*subpage)
-
-            # Save subplot
-            fname = fnm + ".%i." % idx + f
-            plt.savefig(fname, bbox_inches=extent)
-
-            orig_fnm = os.path.join(
-                get_output_dir(parameter.current_set, parameter),
-                parameter.output_file,
-            )
-            fname = orig_fnm + ".%i." % idx + f
-            logger.info(f"Sub-plot saved in: {fname}")
-
-
 def plot(
     parameter: CoreParameter,
     da_test: xr.DataArray,
-    da_ref: xr.DataArray | None,
-    da_diff: xr.DataArray | None,
-    metrics_dict,
```

Comment:
* Updated type annotations for `da_ref` and `da_diff` to make them required. 
* Remove `metrics_dict` because it is unused here and no longer required